provider/keycloak_hardcoded_role_identity_provider_mapper.go

```diff
@@ -0,0 +1,37 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+)
+
+func resourceKeycloakHardcodedRoleIdentityProviderMapper() *schema.Resource {
+	mapperSchema := map[string]*schema.Schema{
+		"role": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Role Name",
+		},
+	}
+	genericMapperResource := resourceKeycloakIdentityProviderMapper()
+	genericMapperResource.Schema = mergeSchemas(genericMapperResource.Schema, mapperSchema)
+	genericMapperResource.Create = resourceKeycloakIdentityProviderMapperCreate(getHardcodedRoleIdentityProviderMapperFromData, setHardcodedRoleIdentityProviderMapperData)
+	genericMapperResource.Read = resourceKeycloakIdentityProviderMapperRead(setHardcodedRoleIdentityProviderMapperData)
+	genericMapperResource.Update = resourceKeycloakIdentityProviderMapperUpdate(getHardcodedRoleIdentityProviderMapperFromData, setHardcodedRoleIdentityProviderMapperData)
+	return genericMapperResource
+}
+
+func getHardcodedRoleIdentityProviderMapperFromData(data *schema.ResourceData, _ interface{}) (*keycloak.IdentityProviderMapper, error) {
+	rec, _ := getIdentityProviderMapperFromData(data)
+	rec.IdentityProviderMapper = "oidc-hardcoded-role-idp-mapper"
```

Comment:
Can this resource not be used for SAML?

Reply:
This resource is also used for SAML, but looks like it's an issue in Keycloak API

Reply:
You can try to create this mapper manually